src/main/java/frc/robot/commands/DriveDistance.java

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.math.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Constants.DriveConstants;
+import frc.robot.subsystems.drive.DriveSubsystemXrp;
+
+public class DriveDistance extends Command {
+  private final DriveSubsystemXrp drive;
+  private final double distance;
+  private final double speed;
+  private final PIDController pidController;
+  private final DifferentialDriveKinematics kinematics;
+  private double rightWheelStartPosition;
+  private double leftWheelStartPosition;
+
+  /**
+   * Creates a new DriveDistance. This command will drive your your robot for a desired distance at
+   * a desired speed.
+   *
+   * @param speed The speed at which the robot will drive
+   * @param meters The number of meters the robot will drive
+   * @param drive The drivetrain subsystem on which this command will run
+   */
+  public DriveDistance(double driveSpeed, double meters, DriveSubsystemXrp driveSubsystem) {
+    distance = meters;
+    speed = driveSpeed;
+    drive = driveSubsystem;
+
+    pidController = new PIDController(1.0, 0.0, 0.0); // Adjust PID constants as needed
+
+    // Create a DifferentialDriveKinematics object with the track width
+    kinematics = new DifferentialDriveKinematics(DriveConstants.trackWidthMeters);
+
+    addRequirements(drive);
```

Comment:
If you change  DriveSubsystemXrp --> Drive, then this needs to be cast to a Subsystem: `addRequirements((Subsystem) drive);`